bus-mapping/src/evm/opcodes/error_invalid_jump.rs

```diff
@@ -0,0 +1,48 @@
+use crate::circuit_input_builder::{CircuitInputStateRef, ExecStep};
+use crate::evm::{Opcode, OpcodeId};
+use crate::Error;
+use eth_types::{GethExecStep, ToAddress, ToWord};
+
+#[derive(Debug, Copy, Clone)]
+pub(crate) struct ErrorInvalidJump;
+
+impl Opcode for ErrorInvalidJump {
+    fn gen_associated_ops(
+        state: &mut CircuitInputStateRef,
+        geth_steps: &[GethExecStep],
+    ) -> Result<Vec<ExecStep>, Error> {
+        let geth_step = &geth_steps[0];
+        let mut exec_step = state.new_step(geth_step)?;
+        let next_step = if geth_steps.len() > 1 {
+            Some(&geth_steps[1])
+        } else {
+            None
+        };
+        exec_step.error = state.get_step_err(geth_step, next_step).unwrap();
+        // assert op code can only be JUMP or JUMPI
+        assert!(geth_step.op == OpcodeId::JUMP || geth_step.op == OpcodeId::JUMPI);
+        let is_jumpi = geth_step.op == OpcodeId::JUMPI;
+        let dest = geth_steps[0].stack.last()?.to_address();
+        let mut condition = 0;
+        if is_jumpi {
+            condition = geth_step.stack.nth_last(1)?.as_usize();
+        }
+        state.stack_read(
+            &mut exec_step,
+            geth_step.stack.last_filled(),
+            dest.to_word(),
+        )?;
+        if is_jumpi {
+            state.stack_read(
+                &mut exec_step,
+                geth_step.stack.nth_last_filled(1),
+                condition.into(),
+            )?;
+        }
+        // `IsSuccess` call context operation is added in gen_restore_context_ops
+
+        state.gen_restore_context_ops(&mut exec_step, geth_steps)?;
```

Comment:
It's confusing that `gen_restore_context_ops` only returns with not generating ops for restoring context when it's error:

https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/4cfccfa6c3b251284ff61eeb907d548d59206753/bus-mapping/src/circuit_input_builder/input_state_ref.rs#L1179-L1190

We should remove this part handle the `IsSuccess` read in `ErrorInvalidJump::gen_associated_ops` and then do:

```suggestion
        if !call.is_root {
            state.gen_restore_context_ops(&mut exec_step, geth_steps)?;
        }
```

Or rename the `gen_restore_context_ops` to a better name.

Reply:
if follow the suggestion, every error case should have the root check codes, reuse it as common helper would be good. I prefer to rename it like `gen_restore_context_ops_if_necessary` , what do you think ? @han0110

Reply:
I see, perhaps we should just keep it as is, we could refactor in a better way in the future.